### PR TITLE
Select last channel viewed

### PR DESCRIPTION
**Pull request by csduarte**
Now saving last channel id to local storage

Reworked local storage saving

### CODE REVIEW

File: app/actions/storage/index.js [hunk 1]
```diff
@@ -24,38 +27,56 @@ export function loadStorage() {
...
+        // TODO: Need to handle this error
+        return null;
+    }
+}
+
+async function saveStorage(data) {
+    try {
+        await AsyncStorage.setItem('storage', JSON.stringify(data));
+    } catch (error) {
+        // TODO: Need to handle this error
```
Comment (enahum) [inline app/actions/storage/index.js]: This should just throw as it will be catched by `updateStorate`

File: app/actions/storage/index.js [hunk 2]
```diff
@@ -3,19 +3,22 @@
...
 export function loadStorage() {
     return async (dispatch, getState) => {
         try {
             const data = JSON.parse(await AsyncStorage.getItem('storage'));
-            const {token, url, currentTeamId} = data;
+
+            const {token, url, currentTeamId, ...otherStorage} = data;
             const credentials = {token, url};
 
+            const currentChannelId = otherStorage[currentTeamId] ? otherStorage[currentTeamId].currentChannelId : '';
```
Comment (hmhealey) [inline app/actions/storage/index.js]: Thoughts on whether or not we should have a separate key for team-specific storage? Like `otherStorage.byTeam[teamId]`? I'm trying to think if we'll ever have naming conflicts here.

Comment (hmhealey) [inline app/actions/storage/index.js]: Yeah, actually I can't come up with any cases where the team id would clash with something else here. Great idea to handle anything team-specific this way!

**Comment by enahum** [comment]
This will need a rebase @jarredwitt 

I'm looking at why is it failing

**Comment by enahum** [comment]
This is ready to be merge once this PR https://github.com/mattermost/platform/pull/5172 is merged in `platform` and we re-run the build for this one

**Pull request merged**